src/Microsoft.ML.SearchSpace/Option/UniformNumericOption.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.SearchSpace.Option
+{
+    public abstract class UniformNumericOption : OptionBase
+    {
+        public UniformNumericOption(double min, double max, bool logBase = false)
+        {
+            Contracts.Check(max > min, "max must be larger than min.");
+            Contracts.Check(min > 0 || logBase == false, "min must be larger than 0 if logBase is true.");
+            this.Min = min;
+            this.Max = max;
+            this.LogBase = logBase;
+            this.Default = Enumerable.Repeat(0.0, this.FeatureSpaceDim).ToArray();
+        }
+
+        public double Min { get; }
+
+        public double Max { get; }
+
+        public bool LogBase { get; }
+
+        public override int FeatureSpaceDim => 1;
+
+        public override int?[] Step => new int?[] { null };
+
+        public override double[] MappingToFeatureSpace(IParameter param)
```

Comment:
Is this code called a lot? Like per each row/cell? Or infrequently?

Reply:
Same question to `SampleFromFeatureSpace`

Reply:
No it's not related with dataset. The two functions (along with `SampleFromFeatureSpace`) here are used to provide a convenient way for tuners to iterating over search space.

See https://github.com/dotnet/machinelearning/pull/6059/files#diff-5ce20f8109d7b89ab236fb1df64594f82530ead3c74d905cb80a9cf0069806c8 for more information

Reply:
### Resolved